src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPiCM3Driver.Linux.cs

```diff
@@ -0,0 +1,598 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Device.Gpio.Drivers
+{
+    public unsafe partial class RaspberryPiCM3Driver : GpioDriver
+    {
+        private const int ENOENT = 2; // error indicates that an entity doesn't exist
```

Comment:
This looks like a lot of copy-pasting. Not something we should be doing.

Reply:
The question is that the two driver (RaspberryP3Driver and RaspberryPiCM3Driver), are exactly the same, except for the number of pins and the pins name. But the [private void ValidatePinNumber(int pinNumber)](https://github.com/dotnet/iot/blob/166f07ac0c7c24605053c12aa6ebc51c0250d256/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs#L17) is private. 
So I had to copy the whole driver, changing only the mentioned two features. Moreover I did not find any guidelines to develop new drivers.
What do you suggest?

Reply:
I'd change ValidatePinNumber to
```
private void ValidatePinNumber(int pinNumber)
        {
            if (pinNumber < 0 || pinNumber >= PinCount)
            {
                throw new ArgumentException("The specified pin number is invalid.", nameof(pinNumber));
            }
        }
```
Hardcoding the 27 there, when a corresponding property exists, seems to be a bug.

Reply:
I agree, In my driver  i changed it in  the same way. But [PinCount](https://github.com/dotnet/iot/blob/166f07ac0c7c24605053c12aa6ebc51c0250d256/src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs#L15), it's not overridable
nor settable

Reply:
The declaration is 
```
protected internal override int PinCount => 28;
```
"protected internal" means protected _or_ internal, so you should be able to override it.

Reply:
Of course, but I cannot find a way to override it and to inherit the whole RaspberryPi3Driver into the new RaspberryPiCM3Driver.